rockcraft/services/package.py

```diff
@@ -64,35 +63,22 @@ def pack(self, prime_dir: pathlib.Path, dest: pathlib.Path) -> list[pathlib.Path
         image_service = services.image
         image_info = image_service.obtain_image()
 
-        platform = self._platform
+        if not self._build_plan:
```

Comment:
Several apps are doing this now. Do you think the upstream PackageService should take a build plan and provide a `_build_info` property that does these checks?

Reply:
I definitely agree that we should upstream this somehow, but in my case the ImageService _also_ needs these checks (and it happens even earlier than the LifecycleService so it's arguable the most important)
Maybe we should expose a "validate_single_build_plan()" function that does these checks and returns a single BuildInfo

Reply:
Feature request made: https://github.com/canonical/craft-application/issues/297

We can figure out details over there; no need to bog down this PR for that.